### PR TITLE
refactor: use aws-sdk-js instead of boto

### PR DESCRIPTION
**Pull request by nornagon**
#### Description of Change
`boto` is deprecated, and `boto3` has substantially more dependencies. We don't currently `pip install` anything during CI machine setup, relying instead on `gclient` to fetch deps. `boto3` has 5 more dependencies. Rather than upgrade to `boto3` and add 5 new manually tracked python dependencies to `DEPS`, switch to using `aws-sdk` in JS to upload files to S3, so we can lean on npm.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] This is **NOT A BREAKING CHANGE**. Breaking changes may not be merged to master until 11-x-y is branched.

#### Release Notes

Notes: none

### CODE REVIEW

File: script/lib/util.py [hunk 1]
```diff
@@ -30,8 +30,7 @@
   os.path.dirname(os.path.dirname(os.path.dirname(__file__)))
 )
 SRC_DIR = os.path.abspath(os.path.join(__file__, '..', '..', '..', '..'))
-BOTO_DIR = os.path.abspath(os.path.join(__file__, '..', '..', '..', 'vendor',
-                                        'boto'))
+BOTO_DIR = os.path.join(ELECTRON_DIR, 'vendor', 'boto')
```
Comment (jkleinsc) [inline script/lib/util.py:33]: Is this needed anymore?

Comment (nornagon) [reply script/lib/util.py]: it was, until i removed the last usage of boto in the next commit

File: script/release/release.js [hunk 2]
```diff
@@ -218,6 +220,36 @@ function uploadIndexJson () {
   console.log(`${pass} Done uploading index.json to S3.`);
 }
 
+async function mergeShasums (pkgVersion) {
+  // Download individual checksum files for Electron zip files from S3,
+  // concatenate them, and upload to GitHub.
+
+  const bucket = process.env.ELECTRON_S3_BUCKET;
+  const accessKey = process.env.ELECTRON_S3_ACCESS_KEY;
+  const secretKey = process.env.ELECTRON_S3_SECRET_KEY;
```
Comment (jkleinsc) [inline script/release/release.js:229]: `secretKey` does not appear to be used.

File: script/lib/s3put.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  file = path.resolve(file);
+  if (file.startsWith(prefix)) file = file.substr(prefix.length);
+  return key_prefix + file.replace(path.sep, '/');
+}
+
+let anErrorOccurred = false;
+function next (done) {
+  const file = files.shift();
+  if (!file) return done();
+  s3.upload({
```
Comment (jkleinsc) [inline script/lib/s3put.js:23]: Where does s3 get authenticated?

File: script/release/release.js (hunk 2 above)
Comment (nornagon) [reply script/release/release.js:229]: good catch! fixed.

File: script/lib/s3put.js (hunk 3 above)
Comment (nornagon) [reply script/lib/s3put.js:23]: It automatically pulls the environment variables: https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/EnvironmentCredentials.html

**Comment by nornagon** [comment]
hm... seeing "access denied" in the logs. I'm not sure if that's due to the environment (since this is a PR and not the normal release flow) or an issue with the code though. @jkleinsc any insights?

**Comment by MarshallOfSound** [comment]
@nornagon Access denied is not expected here, the release builds do have a real token associated with them and should work

**Comment by jkleinsc** [comment]
Merging as CI failure is known flake.

**Pull request merged**